### PR TITLE
MINOR: [CI] update to nightly reports individual job URLs

### PR DESCRIPTION
**Pull request by raulcd**
We currently have around 86K branches on the crossbow repository. This causes some direct problems on browsing and the email reports queries sometimes failing to open.

There are some initiatives to reduce the number of branches (https://github.com/ursacomputing/crossbow/pull/12) both to improve the browsing experience and to avoid possible issues with Github.

One small improvement I propose is to update the URLs on our nightly report emails for individual tasks to avoid a search on github that might timeout.
Instead of using:

https://github.com/ursacomputing/crossbow/branches/all?query=nightly-tests-2022-04-14-0-github-test-build-vcpkg-win

Using the direct branch URL to avoid the search:

https://github.com/ursacomputing/crossbow/tree/nightly-tests-2022-04-14-0-github-test-build-vcpkg-win

### CODE REVIEW

File: dev/archery/archery/crossbow/reports.py [hunk 1]
```diff
@@ -173,10 +173,15 @@ def url(self, query):
         repo_url = self.job.queue.remote_url.strip('.git')
         return '{}/branches/all?query={}'.format(repo_url, query)
 
+    def branch_url(self, branch):
+        repo_url = self.job.queue.remote_url.strip('.git')
```
Comment (pitrou) [inline dev/archery/archery/crossbow/reports.py:177]: This isn't correct:
```python
>>> "something".strip(".git")
'somethin'
```

Starting with Python 3.9 you can use [`str.removesuffix`](https://docs.python.org/3/library/stdtypes.html#str.removesuffix), but for now you'll need something like:
```python
@property
def repo_url(self):
    url = self.job.queue.remote_url
    return url[:-4] if url.endswith('.git') else url
```

**Review by pitrou — CHANGES_REQUESTED** [review]

File: dev/archery/archery/crossbow/reports.py (hunk 1 above)
Comment (raulcd) [reply dev/archery/archery/crossbow/reports.py]: Thanks for that, I did copy&paste from the existing url extraction, I have modified the original one (https://github.com/apache/arrow/blob/master/dev/archery/archery/crossbow/reports.py#L173) to solve that issue too!

**Comment by raulcd** [comment]
I will create a ticket to add tests to archery and crossbow. I have tested the report locally to validate the URLs are correct:
```
$ archery crossbow report --no-fetch nightly-packaging-2022-04-14-0
From: None <None>
To: None
Subject: [NIGHTLY] Arrow Build Report for Job nightly-packaging-2022-04-14-0


Arrow Build Report for Job nightly-packaging-2022-04-14-0

All tasks: https://github.com/raulcd/crossbow/branches/all?query=nightly-packaging-2022-04-14-0

Errored Tasks:
- debian-bookworm-arm64:
  URL: https://github.com/raulcd/crossbow/tree/nightly-packaging-2022-04-14-0-travis-debian-bookworm-arm64
- nuget:
  URL: https://github.com/raulcd/crossbow/tree/nightly-packaging-2022-04-14-0-github-nuget
- wheel-windows-cp310-amd64:
  URL: https://github.com/raulcd/crossbow/tree/nightly-packaging-2022-04-14-0-github-wheel-windows-cp310-amd64
```

**Review by pitrou — APPROVED** [review]
LGTM, thank you @raulcd

**Pull request closed without merging**